interface/output_formatter.py

```diff
@@ -0,0 +1,32 @@
+"""
+This module defines public interface for OutputFormatter class family.
+See test/sample_app.py for example usage.
+"""
+
+import gzip
+from typing import List
+
+from .data_types import OutputData
+
+
+class OutputFormatterInterface:
+
+    def __init__(self, output_name: str, compressed_output: bool) -> None:
```

Comment:
This is exact OOP antipattern.
You should have OutputFormatter instance take "FileLike" argument and let it just do the job of writing to it. Or alternatively, supply "write_output" function to this class. The way it is now, you are mixing two responsibilities in one class hierarchy